Add overview docs (#553)
MicrosoftDocs/quantum-docs-pr: articles/overview/quantum-computers.md

---
title: What can quantum computers do?
description: Learn about the impact of quantum computing, from novel quantum algorithms to quantum inspired algorithms running on classical computers.
author: natke
ms.author: nakersha
ms.date: 10/16/2019
ms.topic: article
uid: microsoft.quantum.overview.computers
---

# What can we do with quantum computers?

What we can do with a quantum computer that can't be done with a classical one?

> Where current computers would require billions of years to solve the world’s most challenging problems, a quantum computer would find a solution in minutes, hours, or days. Quantum computing will enable researchers to develop new catalysts and materials, improve medicines, accelerate advances in artificial intelligence, and answer fundamental questions about the origins of our universe.

Reviewer: ```suggestion
Where current computers would require billions of years to solve the world’s most challenging problems, a quantum computer would find a solution in days, hours, or even minutes. Quantum computing will enable researchers to develop new catalysts and materials, improve medicines, accelerate advances in artificial intelligence, and answer fundamental questions about the origins of our universe.
```

## Molecular simulations

> In a quantum computer, we don’t need to model all of the states of a complex system. Instead the computer itself can be in superposition. We embed the quantum state of the system that we want to simulate in the computer and run the simulation with a specific set of quantum gates. In other words, we use a quantum computer to simulate a quantum system.

Reviewer 2: For the sake of future diffs, I'd suggest breaking this down into one line per sentence.

> In a quantum computer, we don’t need to model all of the states of a complex system. Instead the computer itself can be in superposition. We embed the quantum state of the system that we want to simulate in the computer and run the simulation with a specific set of quantum gates. In other words, we use a quantum computer to simulate a quantum system.

Reviewer: ```suggestion
Simulating microscopic systems that behave according to the laws of quantum mechanics is computationally expensive. We need to take into account all the possible states that can be in superposition and the number of states grows exponentially with the size of the system. In a quantum computer, we don’t need to model all of the states of the system. Instead, we embed the quantum state of the system that we want to simulate in the qubits of the computer itself, and run the simulation with a specific set of quantum gates. In other words, we use a quantum computer to simulate a quantum system.
```

> In a quantum computer, we don’t need to model all of the states of a complex system. Instead the computer itself can be in superposition. We embed the quantum state of the system that we want to simulate in the computer and run the simulation with a specific set of quantum gates. In other words, we use a quantum computer to simulate a quantum system.

Reviewer: ^ basically Eduardo's suggestion but with another change toward the end

> Chemical molecules can be analyzed in this way. One such specific chemical is the `nitrogenase` enzyme, which has the potential to reduce the energy consumption and greenhouse gas emission of current fertilizers.

Reviewer 2: ```suggestion
Chemical molecules can be analyzed in this way. One such specific chemical is the _nitrogenase_ enzyme, which has the potential to reduce the energy consumption and greenhouse gas emission of current fertilizers.
```

> Chemical molecules can be analyzed in this way. One such specific chemical is the _nitrogenase_ enzyme, which has the potential to reduce the energy consumption and greenhouse gas emission of current fertilizers.

Reviewer: ```suggestion
Chemical molecules are quantum systems and therefore can be analyzed in this way. One such specific chemical is the _nitrogenase_ enzyme, which, with a better understanding of its properties, could have the potential to reduce the energy consumption and greenhouse gas emission of current fertilizers. 
```

## Data security and cryptography

Quantum computing is relevant to cryptography in two ways: the first is that classical cryptography relies on the intractability of factorization of large numbers into two prime numbers.

> Quantum computing makes this factorization theoretically tractable (via Shor's algorithm). Whilst implementation of this algorithm is not physically possible with the current scale of quantum hardware, it has spawned development of quantum resistant algorithms to future-proof data security, including novel quantum algorithms for encryption and cryptographic key distribution.

Reviewer: Correct me if I'm wrong @natke, but I think the two ways you were getting at are 
1) uh oh, quantum computing can break some of the most common classical encryption methods
2) the development of new security protocols, some of which are classical and quantum-resistant, and others that actually make use of quantum-ness itself (e.g. quantum key distribution).

I'd say just re-group the paragraphs to match that structure (first sentence of paragraph 2 bumped up to paragraph 1), and then throw in wording akin to my (1) and (2) above (i.e. no need to actually mention discrete logs/elliptic crypt by name, and expand the second paragraph with a sentence or two)

## Quantum-inspired computing

Quantum-inspired algorithms are implemented with classical software, but use quantum principles for increased speed and accuracy.

Quantum-inspired algorithms are being applied to medical research. For example, to improve the accuracy of Magnetic Resonance Imaging (MRI) scans. Quantum-inspired computing is being used to optimize the configuration of the MRI machines for identification of specific diseases.

> 

Reviewer: ^ a good point. Perhaps just make it the last section on the page, and then contextualize it a bit: 
"research on quantum algorithms hasn't just helped us better understand 'what quantum computers can do', but also lent invaluable insight to classical computing" 
... or something similar

## Big data and machine learning

### Search

> By exploiting superposition, Grover's algorithm only needs O(√N) queries to find a specific item in a database, rather than having to access every item. It means that if we have a database of, for example, one million elements, we only need around 1000 tries to find the target.

Reviewer 2: ```suggestion
By exploiting superposition, Grover's algorithm only needs $O(\sqrt{N})$ queries to find a specific item in a database, rather than having to access every item. It means that if we have a database of, for example, one million elements, we only need around 1000 tries to find the target.
```

### Solving linear systems of equations

A great number of numerical calculations on classical computing consist mainly in solving linear systems of equations. This is especially true in the field of machine learning where most of the computation cost goes into calculating the inverse of huge matrices.

Fortunately, there exists a quantum algorithm that allows us to approximately solve the system exponentially faster than a classical computer. This opens the door to great speedups in every problem that needs the solution to linear systems of equations.

## Dive deeper

> You can learn how to implement some of these quantum algorithms with the [Quantum Katas](~/intro-to-katas.md)

Reviewer 2: Path vs xref.

Author: Fixed

> You can learn how to implement some of these quantum algorithms with the [Quantum Katas](xref:microsoft.quantum.overview.katas)

Reviewer: Good spot to link to the Grover quickstart @cgranade is working on? (when it's done of course)
